File: solutions/fortran/darts/darts.f90
module darts
   implicit none

contains

   pure integer function score(x, y) result(points)
      real, intent(in):: x, y
      real :: dist ! distance from center
      dist = sqrt(x * x + y * y)

      if (dist <= 1) then
         ! in the inner circle
         points = 10
      elseif (dist <= 5) then
         ! in the middle circle
         points = 5
      elseif (dist <= 10) then
         ! in the outer circle
         points = 1
      else
         ! outside the target
         points = 0
      end if
   end function score

end module darts

File: solutions/go/darts/darts.go
package darts

import "math"

func Score(x, y float64) int {
	d := distance(x, y)
	switch {
	case d <= 1:
		return 10
	case d <= 5:
		return 5
	case d <= 10:
		return 1
	default:
		return 0
	}
}

func distance(x, y float64) float64 {
	return math.Sqrt(x*x + y*y)
}
